x-pack/plugins/ml/public/application/components/ml_saved_objects_spaces_list/ml_saved_objects_spaces_list.tsx

```diff
@@ -19,7 +19,7 @@ import { useMlApiContext } from '../../contexts/kibana';
 import { useToastNotificationService } from '../../services/toast_notification_service';
 
 interface Props {
-  spacesApi: SpacesPluginStart;
+  spacesApi: SpacesPluginStart; // this component is only ever used when spaces is enabled
```

Comment:
should we make it optional then?

Reply:
It depends how you want to break up the components. 

You could have a component that requires `spacesApi` prop but then the parent component that renders it would have to show/hide it depending on whether `spacesApi` is available. 

It has to be optional at the very root level of your render tree but the rest is up to you.